test/acceptance/user-order.controller.acceptance.ts

```diff
@@ -0,0 +1,199 @@
+
+  before(setupApplication);
+  beforeEach(clearDatabase);
+  after(async () => {
+    await app.stop();
+  });
+
+  it('creates an order for a user with a given orderId', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = {
+      orderId: '1',
+      userId: userId,
+      total: 99.99,
+      products: [
+        {
+          productId: 'product1',
+          quantity: 10,
+          price: 9.99,
+        },
+      ],
+    };
+
+    await client
+      .post(`/users/${userId}/orders`)
+      .send(order)
+      .expect(200, order);
+  });
+
+  it('creates an order for a user without a given orderId', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = givenAOrder();
+    order.userId = userId;
+
+    const res = await client
+      .post(`/users/${userId}/orders`)
+      .send(order)
+      .expect(200);
+
+    expect(res.body.orderId).to.be.a.String();
+    delete res.body.orderId;
+    expect(res.body).to.deepEqual(JSON.parse(JSON.stringify(order)));
+  });
+
+  it('throws an error when a userId in path does not match body', async () => {
+    const user = await givenAUser();
+    const userId = user.id.toString();
+    const order = givenAOrder();
+    order.userId = 'hello';
+
+    await client
+      .post(`/users/${userId}/orders`)
+      .set('Content-Type', 'application/json')
+      .send(order)
+      .expect(400);
+  });
+
+  it.skip('throws an error when creating an order for a non-existent user', async () => {});
+
+  describe('with multiple orders', () => {
+    let user: User;
+    let userId: String;
+    let order1: Partial<Order>;
+    let order2: Partial<Order>;
+    let savedOrder1: Order;
+    let savedOrder2: Order;
+
+    beforeEach(givenUserAndOrders);
+
+    it('retrieves orders for a given user', async () => {
+      const order = givenAOrder();
+      order.userId = 'randomUserId';
+      order.total = 100.99;
+      await orderRepo.create(order);
+
+      const expected = JSON.parse(JSON.stringify([savedOrder1, savedOrder2]));
+
+      await client.get(`/users/${userId}/orders`).expect(200, expected);
+    });
+
+    it('patches all orders for a given user', async () => {
+      await client
+        .patch(`/users/${userId}/orders`)
+        .send({total: 9.99})
+        .expect(200, '2');
+
+      const getOrder1 = await orderRepo.find({
+        where: {orderId: savedOrder1.orderId},
+      });
+      savedOrder1.total = 9.99;
+      savedOrder1.orderId = savedOrder1.orderId!.toString();
+      expect(getOrder1).to.deepEqual([savedOrder1]);
+
+      const getOrder2 = await orderRepo.find({
+        where: {orderId: savedOrder2.orderId},
+      });
+      savedOrder2.total = 9.99;
+      savedOrder2.orderId = savedOrder2.orderId!.toString();
+      expect(getOrder2).to.deepEqual([savedOrder2]);
+    });
+
+    it.skip('patches orders matching filter for a given user', async () => {});
```

Comment:
Ditto.

Please create a follow-up issue (if you haven't do so already) and add a code comment pointing to that issue. (Explain that this test is skipped because the linked issue is not implemented yet.)